compiler/rustc_mir/src/const_eval/machine.rs

```diff
@@ -43,10 +43,25 @@ impl<'mir, 'tcx> InterpCx<'mir, 'tcx, CompileTimeInterpreter<'mir, 'tcx>> {
             let msg = Symbol::intern(self.read_str(&msg_place)?);
             let span = self.find_closest_untracked_caller_location();
             let (file, line, col) = self.location_triple_for_span(span);
-            Err(ConstEvalErrKind::Panic { msg, file, line, col }.into())
-        } else {
-            Ok(())
+            return Err(ConstEvalErrKind::Panic { msg, file, line, col }.into());
+        } else if Some(def_id) == self.tcx.lang_items().panic_fmt()
+            || Some(def_id) == self.tcx.lang_items().begin_panic_fmt()
+        {
+            // For panic_fmt, call const_panic_fmt instead.
```

Comment:
FWIW, this is an ad-hoc implementation of "call a different function at runtime vs compiletime".

Reply:
Yup. If a generic mechanism for that ever happens, that can be used instead.